src/components/vm/consoles/consoles.jsx

```diff
@@ -81,15 +84,42 @@ export class ConsoleCardState extends StateObject {
     }
 }
 
-export const ConsoleCard = ({ state, vm, config, onAddErrorNotification, isExpanded }) => {
+export const ConsoleCard = ({ state, vm, config, onAddErrorNotification, isExpanded, isStandalone }) => {
     const serials = vm.displays.filter(display => display.type == 'pty');
     const inactive_serials = vm.inactiveXML.displays.filter(display => display.type == 'pty');
     const vnc = vm.displays.find(display => display.type == 'vnc');
     const inactive_vnc = vm.inactiveXML.displays.find(display => display.type == 'vnc');
     const spice = vm.displays.find(display => display.type == 'spice');
     const inactive_spice = vm.inactiveXML.displays.find(display => display.type == 'spice');
 
-    let type = state.type;
+    const lastVncRemoteSize = useRef([1024, 768]);
+
+    const [globallyConnected, setGloballyConnected] = useState(true);
+
+    useEffect(() => {
+        if (isStandalone) {
+            const watchdog = cockpit.channel({ payload: "null" });
+            watchdog.addEventListener("close", () => {
+                setGloballyConnected(false);
+            });
+            return () => watchdog.close();
+        }
+    }, [isStandalone]);
+
+    if (!globallyConnected) {
+        return (
+            <EmptyState>
+                <EmptyStateBody>{_("Session closed")}</EmptyStateBody>
+                <EmptyStateFooter>
+                    <Button variant="primary" onClick={() => window.location.reload()}>
+                        {_("Login")}
+                    </Button>
+                </EmptyStateFooter>
+            </EmptyState>
```

Comment:
These 10 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2254-839cfa62-20250805-072522-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)